### ISSUE TITLE
IDEA 14 Time to time error occurred when format Java class.

### ISSUE THREAD

**Issue by AManuev** [krasa/EclipseCodeFormatter#33]
Hi,  
Exception log:

Already disposed: Project (Disposed) xxxxxxx
java.lang.AssertionError: Already disposed: Project (Disposed) xxxxxx
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getComponent(ComponentManagerImpl.java:191)
	at com.intellij.psi.PsiManager.getInstance(PsiManager.java:39)
	at com.intellij.psi.impl.smartPointers.SelfElementInfo$1.compute(SelfElementInfo.java:238)
	at com.intellij.psi.impl.smartPointers.SelfElementInfo$1.compute(SelfElementInfo.java:224)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:917)
	at com.intellij.psi.impl.smartPointers.SelfElementInfo.restoreFileFromVirtual(SelfElementInfo.java:224)
	at com.intellij.psi.impl.smartPointers.SelfElementInfo.restoreFile(SelfElementInfo.java:174)
	at com.intellij.psi.impl.smartPointers.AnchorElementInfo.restoreElement(AnchorElementInfo.java:62)
	at com.intellij.psi.impl.smartPointers.AnchorElementInfo.fastenBelt(AnchorElementInfo.java:105)
	at com.intellij.psi.impl.smartPointers.SmartPsiElementPointerImpl.fastenBelt(SmartPsiElementPointerImpl.java:177)
	at com.intellij.psi.impl.smartPointers.SmartPointerManagerImpl.fastenBelts(SmartPointerManagerImpl.java:67)
	at com.intellij.psi.impl.PsiDocumentManagerBase.beforeDocumentChange(PsiDocumentManagerBase.java:614)
	at sun.reflect.GeneratedMethodAccessor208.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.intellij.util.EventDispatcher.dispatch(EventDispatcher.java:88)
	at com.intellij.util.EventDispatcher.access$100(EventDispatcher.java:34)
	at com.intellij.util.EventDispatcher$1.invoke(EventDispatcher.java:68)
	at com.sun.proxy.$Proxy19.beforeDocumentChange(Unknown Source)
	at com.intellij.openapi.editor.impl.DocumentImpl.a(DocumentImpl.java:746)
	at com.intellij.openapi.editor.impl.DocumentImpl.a(DocumentImpl.java:714)
	at com.intellij.openapi.editor.impl.DocumentImpl.a(DocumentImpl.java:601)
	at com.intellij.openapi.editor.impl.DocumentImpl.access$400(DocumentImpl.java:57)
	at com.intellij.openapi.editor.impl.DocumentImpl$7.run(DocumentImpl.java:985)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:117)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:99)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:85)
	at com.intellij.openapi.editor.impl.DocumentImpl.setText(DocumentImpl.java:992)
	at krasa.formatter.plugin.EclipseCodeFormatter.formatWhenEditorIsOpen(EclipseCodeFormatter.java:84)
	at krasa.formatter.plugin.EclipseCodeFormatter.format(EclipseCodeFormatter.java:50)
	at krasa.formatter.plugin.EclipseCodeStyleManager.formatWithEclipse(EclipseCodeStyleManager.java:189)
	at krasa.formatter.plugin.EclipseCodeStyleManager.format(EclipseCodeStyleManager.java:95)
	at krasa.formatter.plugin.EclipseCodeStyleManager.reformatText(EclipseCodeStyleManager.java:59)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor$1.call(ReformatCodeProcessor.java:95)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor$1.call(ReformatCodeProcessor.java:89)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:206)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:198)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:202)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:198)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$3.run(AbstractLayoutCodeProcessor.java:307)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:974)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$8$1$1.run(AbstractLayoutCodeProcessor.java:415)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:124)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:99)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:85)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$8$1.run(AbstractLayoutCodeProcessor.java:410)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:318)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:744)
	at java.awt.EventQueue.access$400(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:697)
	at java.awt.EventQueue$3.run(EventQueue.java:691)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:714)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:577)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

**Comment by krasa** [comment]
I've seen a lot of similar ones, never with this plugin though... It should be fixed in some new version of IJ.

```
Report #773733
Fixed in build 140.508


Already disposed: Project (Disposed) SVNMergeInfo


java.lang.AssertionError: Already disposed: Project (Disposed) SVNMergeInfo
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getComponent(ComponentManagerImpl.java:191)
	at com.intellij.psi.PsiManager.getInstance(PsiManager.java:39)
	at com.intellij.psi.impl.smartPointers.SelfElementInfo$1.compute(SelfElementInfo.java:238)
	at com.intellij.psi.impl.smartPointers.SelfElementInfo$1.compute(SelfElementInfo.java:224)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:920)
	at com.intellij.psi.impl.smartPointers.SelfElementInfo.restoreFileFromVirtual(SelfElementInfo.java:224)
	at com.intellij.psi.impl.smartPointers.SelfElementInfo.restoreFile(SelfElementInfo.java:174)
	at com.intellij.psi.impl.smartPointers.AnchorElementInfo.restoreElement(AnchorElementInfo.java:62)
	at com.intellij.psi.impl.smartPointers.AnchorElementInfo.fastenBelt(AnchorElementInfo.java:105)
	at com.intellij.psi.impl.smartPointers.SmartPsiElementPointerImpl.fastenBelt(SmartPsiElementPointerImpl.java:177)
	at com.intellij.psi.impl.smartPointers.SmartPointerManagerImpl.fastenBelts(SmartPointerManagerImpl.java:67)
	at com.intellij.psi.impl.PsiDocumentManagerBase.beforeDocumentChange(PsiDocumentManagerBase.java:614)
	at sun.reflect.GeneratedMethodAccessor112.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.util.EventDispatcher.dispatch(EventDispatcher.java:88)
	at com.intellij.util.EventDispatcher.access$100(EventDispatcher.java:34)
	at com.intellij.util.EventDispatcher$1.invoke(EventDispatcher.java:68)
	at com.sun.proxy.$Proxy22.beforeDocumentChange(Unknown Source)
	at com.intellij.openapi.editor.impl.DocumentImpl.doBeforeChangedUpdate(DocumentImpl.java:746)
	at com.intellij.openapi.editor.impl.DocumentImpl.updateText(DocumentImpl.java:714)
	at com.intellij.openapi.editor.impl.DocumentImpl.insertString(DocumentImpl.java:498)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.doCommitTransaction(PsiToDocumentSynchronizer.java:287)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.access$100(PsiToDocumentSynchronizer.java:37)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer$6.syncDocument(PsiToDocumentSynchronizer.java:262)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer$1.run(PsiToDocumentSynchronizer.java:102)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.performAtomically(PsiToDocumentSynchronizer.java:126)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.doSync(PsiToDocumentSynchronizer.java:99)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.commitTransaction(PsiToDocumentSynchronizer.java:259)
	at com.intellij.pom.core.impl.PomModelImpl.commitTransaction(PomModelImpl.java:235)
	at com.intellij.pom.core.impl.PomModelImpl.runTransaction(PomModelImpl.java:194)
	at com.intellij.psi.impl.source.tree.ChangeUtil.prepareAndRunChangeAction(ChangeUtil.java:165)
	at com.intellij.psi.impl.source.tree.CompositeElement.addChild(CompositeElement.java:654)
	at com.intellij.psi.impl.source.tree.CompositeElement.addChildren(CompositeElement.java:761)
	at com.intellij.psi.impl.source.codeStyle.CodeEditUtil.addChildren(CodeEditUtil.java:82)
	at com.intellij.psi.impl.source.tree.CompositeElement.addInternal(CompositeElement.java:540)
	at com.intellij.psi.impl.source.tree.java.PsiCodeBlockImpl.addInternal(PsiCodeBlockImpl.java:161)
	at com.intellij.psi.impl.source.tree.LazyParseablePsiElement.addInnerBefore(LazyParseablePsiElement.java:342)
	at com.intellij.psi.impl.source.tree.LazyParseablePsiElement.addBefore(LazyParseablePsiElement.java:179)
	at com.intellij.refactoring.introduceVariable.IntroduceVariableBase$5.introduce(IntroduceVariableBase.java:946)
	at com.intellij.refactoring.introduceVariable.IntroduceVariableBase$5.run(IntroduceVariableBase.java:864)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:977)
	at com.intellij.refactoring.introduceVariable.IntroduceVariableBase$3$1.run(IntroduceVariableBase.java:694)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:124)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:99)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:85)
	at com.intellij.refactoring.introduceVariable.IntroduceVariableBase$3.pass(IntroduceVariableBase.java:689)
	at com.intellij.refactoring.introduceVariable.IntroduceVariableBase$3.pass(IntroduceVariableBase.java:645)
	at com.intellij.refactoring.introduce.inplace.OccurrencesChooser.showChooser(OccurrencesChooser.java:93)
	at com.intellij.refactoring.introduceVariable.IntroduceVariableBase.invokeImpl(IntroduceVariableBase.java:727)
	at com.intellij.refactoring.introduceVariable.IntroduceVariableBase.invoke(IntroduceVariableBase.java:268)
	at com.intellij.refactoring.introduceVariable.IntroduceVariableBase.invoke(IntroduceVariableBase.java:146)
	at com.intellij.refactoring.actions.BaseRefactoringAction.actionPerformed(BaseRefactoringAction.java:122)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$3.performAction(IdeKeyEventDispatcher.java:614)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:663)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.java:513)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:211)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:546)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
```

**Comment by krasa** [comment]
```
Already disposed
java.lang.AssertionError: Already disposed
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:281)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:53)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:40)
	at com.intellij.codeStyle.CodeStyleFacade.getInstance(CodeStyleFacade.java:38)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl.a(LazyRangeMarkerFactoryImpl.java:308)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl.access$300(LazyRangeMarkerFactoryImpl.java:38)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl$LineColumnLazyMarker.createDelegate(LazyRangeMarkerFactoryImpl.java:274)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl$LazyMarker.getOrCreateDelegate(LazyRangeMarkerFactoryImpl.java:158)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl$1.a(LazyRangeMarkerFactoryImpl.java:71)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl$1.beforeDocumentChange(LazyRangeMarkerFactoryImpl.java:48)
	at sun.reflect.GeneratedMethodAccessor140.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.util.EventDispatcher.dispatch(EventDispatcher.java:88)
	at com.intellij.util.EventDispatcher.access$100(EventDispatcher.java:34)
	at com.intellij.util.EventDispatcher$1.invoke(EventDispatcher.java:68)
	at com.sun.proxy.$Proxy22.beforeDocumentChange(Unknown Source)
	at com.intellij.openapi.editor.impl.DocumentImpl.a(DocumentImpl.java:746)
	at com.intellij.openapi.editor.impl.DocumentImpl.a(DocumentImpl.java:714)
	at com.intellij.openapi.editor.impl.DocumentImpl.deleteString(DocumentImpl.java:522)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.a(PsiToDocumentSynchronizer.java:284)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.access$100(PsiToDocumentSynchronizer.java:37)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer$6.syncDocument(PsiToDocumentSynchronizer.java:262)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer$1.run(PsiToDocumentSynchronizer.java:102)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.performAtomically(PsiToDocumentSynchronizer.java:126)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.a(PsiToDocumentSynchronizer.java:99)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.commitTransaction(PsiToDocumentSynchronizer.java:259)
	at com.intellij.pom.core.impl.PomModelImpl.a(PomModelImpl.java:235)
	at com.intellij.pom.core.impl.PomModelImpl.runTransaction(PomModelImpl.java:194)
	at com.intellij.psi.impl.source.tree.ChangeUtil.prepareAndRunChangeAction(ChangeUtil.java:165)
	at com.intellij.psi.impl.source.tree.CompositeElement.replaceChild(CompositeElement.java:703)
	at com.intellij.psi.impl.source.codeStyle.CodeEditUtil.replaceChild(CodeEditUtil.java:205)
	at com.intellij.psi.impl.source.tree.CompositeElement.replaceChildInternal(CompositeElement.java:548)
	at com.intellij.psi.impl.source.tree.java.JavaFileElement.replaceChildInternal(JavaFileElement.java:97)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.doReplace(SharedImplUtil.java:207)
	at com.intellij.psi.impl.source.JavaStubPsiElement.replace(JavaStubPsiElement.java:150)
	at com.intellij.lang.java.JavaImportOptimizer$1.run(JavaImportOptimizer.java:58)
	at krasa.formatter.plugin.EclipseImportOptimizer$1.run(EclipseImportOptimizer.java:40)
	at com.intellij.codeInsight.actions.OptimizeImportsProcessor$1.run(OptimizeImportsProcessor.java:86)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:206)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:198)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ReformatFilesTask.a(AbstractLayoutCodeProcessor.java:511)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ReformatFilesTask.iteration(AbstractLayoutCodeProcessor.java:502)
	at com.intellij.util.SequentialModalProgressTask$1.run(SequentialModalProgressTask.java:95)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:302)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:733)
	at java.awt.EventQueue.access$200(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:694)
	at java.awt.EventQueue$3.run(EventQueue.java:692)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:703)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:577)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
	at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:832)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:208)
	at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:660)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.a(ProgressManagerImpl.java:418)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.run(ProgressManagerImpl.java:578)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$4.run(AbstractLayoutCodeProcessor.java:336)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$6.run(AbstractLayoutCodeProcessor.java:355)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:977)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$8$1$1.run(AbstractLayoutCodeProcessor.java:415)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:124)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:99)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:85)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$8$1.run(AbstractLayoutCodeProcessor.java:410)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:318)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:312)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:733)
	at java.awt.EventQueue.access$200(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:694)
	at java.awt.EventQueue$3.run(EventQueue.java:692)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:703)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:577)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
```


------------


```
Already disposed
java.lang.AssertionError: Already disposed
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:281)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:53)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:40)
	at com.intellij.codeStyle.CodeStyleFacade.getInstance(CodeStyleFacade.java:38)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl.a(LazyRangeMarkerFactoryImpl.java:308)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl.access$300(LazyRangeMarkerFactoryImpl.java:38)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl$LineColumnLazyMarker.createDelegate(LazyRangeMarkerFactoryImpl.java:274)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl$LazyMarker.getOrCreateDelegate(LazyRangeMarkerFactoryImpl.java:158)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl$1.a(LazyRangeMarkerFactoryImpl.java:71)
	at com.intellij.openapi.editor.impl.LazyRangeMarkerFactoryImpl$1.beforeDocumentChange(LazyRangeMarkerFactoryImpl.java:48)
	at sun.reflect.GeneratedMethodAccessor140.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.util.EventDispatcher.dispatch(EventDispatcher.java:88)
	at com.intellij.util.EventDispatcher.access$100(EventDispatcher.java:34)
	at com.intellij.util.EventDispatcher$1.invoke(EventDispatcher.java:68)
	at com.sun.proxy.$Proxy22.beforeDocumentChange(Unknown Source)
	at com.intellij.openapi.editor.impl.DocumentImpl.a(DocumentImpl.java:746)
	at com.intellij.openapi.editor.impl.DocumentImpl.a(DocumentImpl.java:714)
	at com.intellij.openapi.editor.impl.DocumentImpl.a(DocumentImpl.java:601)
	at com.intellij.openapi.editor.impl.DocumentImpl.access$400(DocumentImpl.java:57)
	at com.intellij.openapi.editor.impl.DocumentImpl$7.run(DocumentImpl.java:985)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:117)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:99)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:85)
	at com.intellij.openapi.editor.impl.DocumentImpl.setText(DocumentImpl.java:992)
	at krasa.formatter.plugin.EclipseCodeFormatter.formatWhenEditorIsClosed(EclipseCodeFormatter.java:67)
	at krasa.formatter.plugin.EclipseCodeFormatter.format(EclipseCodeFormatter.java:52)
	at krasa.formatter.plugin.EclipseCodeStyleManager.formatWithEclipse(EclipseCodeStyleManager.java:189)
	at krasa.formatter.plugin.EclipseCodeStyleManager.format(EclipseCodeStyleManager.java:95)
	at krasa.formatter.plugin.EclipseCodeStyleManager.reformatText(EclipseCodeStyleManager.java:59)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor$1.call(ReformatCodeProcessor.java:95)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor$1.call(ReformatCodeProcessor.java:89)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:206)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:198)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:202)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$1.call(AbstractLayoutCodeProcessor.java:198)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ReformatFilesTask.a(AbstractLayoutCodeProcessor.java:511)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ReformatFilesTask.iteration(AbstractLayoutCodeProcessor.java:502)
	at com.intellij.util.SequentialModalProgressTask$1.run(SequentialModalProgressTask.java:95)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:302)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:733)
	at java.awt.EventQueue.access$200(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:694)
	at java.awt.EventQueue$3.run(EventQueue.java:692)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:703)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:577)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
	at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:832)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:208)
	at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:660)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.a(ProgressManagerImpl.java:418)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.run(ProgressManagerImpl.java:578)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$4.run(AbstractLayoutCodeProcessor.java:336)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$6.run(AbstractLayoutCodeProcessor.java:355)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:977)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$8$1$1.run(AbstractLayoutCodeProcessor.java:415)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:124)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:99)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:85)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$8$1.run(AbstractLayoutCodeProcessor.java:410)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:318)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:312)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:733)
	at java.awt.EventQueue.access$200(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:694)
	at java.awt.EventQueue$3.run(EventQueue.java:692)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:703)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:577)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
```

**Comment by krasa** [comment]
seems fixed in IJ 14.1
